Implement UI for channel page

rhysyngsun opened the pull request:
#### What are the relevant tickets?
Part of #26 

#### What's this PR do?
Adds the channel listing page.

#### How should this be manually tested?
Go to /channels/something and verify you see the page as shown in the screenshot.
Also go to `/api/v0/channels` and `/api/v0/channels/pics` and verify you see `public_description` in there now. it will probably be empty strings unless you've set it on a sub.

#### Screenshots (if appropriate)
![selection_077](https://user-images.githubusercontent.com/28598/28482057-bbcddca4-6e34-11e7-8a68-71777770ac73.png)

#### What GIF best describes this PR or how it makes you feel?
![](https://media.giphy.com/media/8YgBbvqazclxu/giphy.gif)

aliceriot commented:
If I go to `/channels/foobar` or `/channels/test_channel` I just see a blank page. It looks like the `Route` component is rendering `null` for it's children somehow?

![nullnullwhy](https://user-images.githubusercontent.com/6207644/28527423-6f443ce0-7058-11e7-883c-31a5daa15936.png)

aliceriot commented:
I also don't seem to be able to visit the API pages (like `/api/v0/channels/pics`), I just get an empty React app at that URL.

rhysyngsun commented:
@aliceriot I've fixed those. One was a regression with how I dealt with urls.py, the other was the new `<Loading/>` I wrote in this PR was expecting the `posts` reducer to set `loaded = true` which it never did. I removed that from the list for now.

aliceriot commented:
API stuff appears to be working! although we should probably note down the quirk that new users have to take some sort of 'action' (subscribing to a subreddit worked for me) before their subscriptions show up in the API.

aliceriot commented:
UI shows up too

rhysyngsun commented:
@aliceriot I'll file a bug for the subreddit list not showing up right away. Might be some user initialization that we need to add to the refresh_token api. CC @noisecapella

aliceriot commented on static/scss/sidebar.scss:
```diff
@@ -0,0 +1,20 @@
+
+
```
blank lines

rhysyngsun replied:
:trollface:
